vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -1138,7 +1032,66 @@ public void scrollToIndex(int... indexes) {
     public void scrollToEnd() {
         getUI().ifPresent(ui -> ui.beforeClientResponse(this,
                 ctx -> getElement().executeJs(
-                        "this.scrollToIndex(...Array(10).fill(Infinity))")));
+                        "this.scrollToIndex(...Array(10).fill(-1))")));
+    }
+
+    /**
+     * Sets the viewport range centered on the item specified by a hierarchical
+     * index path e.g. { 0, 1, 1 }. The {@code padding} parameter specifies how
+     * many items should be added on each side of the center item in the
+     * resulting range.
+     * <p>
+     * This method has package-private visibility to allow testing.
+     *
+     * @param path
+     *            the path to the item to use as the center of the viewport
+     *            range
+     * @param padding
+     *            the number of items to add on each side of the center item
+     */
+    @ClientCallable
```

Comment:
Do we need `@AllowInert` for this `@ClientCallable` or it's not supposed to be called in the background?

Reply:
Good question. Considering that `setViewportRange` has `@AllowInert`, I guess this method should also have it.

Reply:
It would be good to verify it with `InertTreeGridIT` added in https://github.com/vaadin/flow-components/pull/6465.

Reply:
I think it would make sense to add it. The main purpose of the method is to be called through the server itself with `scrollToIndex` which due to the design requires a roundtrip to the client. It also shouldn't cause any harm otherwise, as it essentially does the same thing as loading data which is already allowed when inert.